### ISSUE TITLE
What path functions are you using?

### ISSUE THREAD

**Issue by vinkla** [wordplate/framework#79]
WordPlate's goal has always been to enhance the developer experience with WordPress. Making it easier to work together with WordPress with newer standards in PHP while not adding too many custom features.

I've recently looked through which path functions we've implemented in WordPlate which works similarly to functions in WordPress. With this issue I'm raising the question if it be possible to deprecate our custom path functions and use WordPress functions instead.

If you've a minute to spare, please answer these questions about WordPlate's path functions:

1. Are you using the `base_path` function?
1. Are you using the `template_path` function?
1. Are you using the `stylesheet_path` function?
1. When do you use `stylesheet_path` instead of `template_path`?
1. When do you use `template_path` and/or `stylesheet_path` instead of [`get_theme_file_path`](https://developer.wordpress.org/reference/functions/get_theme_file_path/)?

cc @wordplate/developers

**Comment by puredazzle** [comment]
1. No
2. A lot, for example `require template_path('components/product-page/search.php');`
3. No
4. -
5. I use template_path because of old habits I guess.

**Comment by fiskhandlarn** [comment]
> 1. Are you using the base_path function?

Yes. For including vendor-plugins:
`require base_path('vendor/johnbillion/extended-cpts/extended-cpts.php');`

But maybe I can use `get_theme_file_path` like this instead?
`require get_theme_file_path('../../../vendor/johnbillion/extended-cpts/extended-cpts.php');`


> 2. Are you using the template_path function?

Yes, in my parent theme.


> 3. Are you using the stylesheet_path function?

Yes, in my child theme.


> 4. When do you use stylesheet_path instead of template_path?

When including files in the child theme.


> 5. When do you use template_path and/or stylesheet_path instead of get_theme_file_path?

Always until now. But if I were to begin using `get_theme_file_path` I would still need to use `template_path` in my parent theme (since `get_theme_file_path` defaults to files in the child theme if found).

Please keep these helpers! X) (I guess I could do without `stylesheet_path` and use `get_theme_file_path` instead.)

**Comment by vinkla** [comment]
> `require base_path('vendor/johnbillion/extended-cpts/extended-cpts.php');`

I guess in this case you could [rely on Composer instead](https://github.com/johnbillion/extended-cpts/blob/fb48e7905d81796ac9cf597988a0a52593356ad8/composer.json#L51-L55)?


> Please keep these helpers! X) (I guess I could do without stylesheet_path and use get_theme_file_path instead.)

😏👍

**Comment by fiskhandlarn** [comment]
> > `require base_path('vendor/johnbillion/extended-cpts/extended-cpts.php');`
> 
> I guess in this case you could [rely on Composer instead](https://github.com/johnbillion/extended-cpts/blob/fb48e7905d81796ac9cf597988a0a52593356ad8/composer.json#L51-L55)?

Fuck yeah, thanks! ^^

**Comment by fiskhandlarn** [comment]
@vinkla I had to write it like this though:

```
  "autoload": {
    "files": [
      "vendor/johnbillion/extended-cpts/extended-cpts.php"
    ]
  },
```

**Comment by vinkla** [comment]
@fiskhandlarn hmm, how do you install the package? Since you're autoloading the same file as `johnbillion/extended-cpts` you shouldn't need to autoload it yourself.

**Comment by fiskhandlarn** [comment]
@vinkla Silly me, the package is of course already included, NM! X)

**Comment by vinkla** [comment]
This is now merged with #81.
